src/plugins/intel_gna/src/gna_plugin_config.cpp

```diff
@@ -46,6 +46,39 @@ static const caseless_unordered_map<std::string, std::pair<Gna2AccelerationMode,
 };
 OPENVINO_SUPPRESS_DEPRECATED_END
 
+/**
+ * @brief Parse string into value without throwing
+ * @param value A parsed value
+ * @param def_value A value to return in case of exception
+ * @return parsed value, default value otherwise
+ * @ingroup ov_runtime_gna_prop_cpp_api
+ */
+template <class T>
+static T FromString(const std::string& value, T def_value) {
+    T result = def_value;
+    try {
+        result = ov::util::from_string<T>(value);
+    } catch (std::exception&) {
+    }
+    return result;
+};
+
+/**
+ * @brief Update the gnaPrecision basing on execution_mode and inference_precision values
+ * @ingroup ov_runtime_gna_prop_cpp_api
+ */
+void Config::UpdateGNAPrecision() {
+    if (inference_precision == ov::element::undefined) {
+        if (execution_mode != ov::hint::ExecutionMode::UNDEFINED) {
+            gnaPrecision = (execution_mode == ov::hint::ExecutionMode::PERFORMANCE) ? Precision::I8 : Precision::I16;
+        }
+    } else {
+        gnaPrecision = (inference_precision == ov::element::i8) ? Precision::I8 : Precision::I16;
+        execution_mode = (inference_precision == ov::element::i8) ? ov::hint::ExecutionMode::PERFORMANCE
```

Comment:
We also modify execution_mode here.
I am not sure if it will be clear for everyone based on function name or its description that inference_precision has higher priority and may override execution_mode

Reply:
Added more explanatory description and changed method name.